core/field_variable.ts

```diff
@@ -89,9 +88,7 @@ export class FieldVariable extends FieldDropdown {
      * An array of options for a dropdown list,
      * or a function which generates these options.
      */
-    // AnyDuringMigration because:  Type '(this: FieldVariable) => any[][]' is
-    // not assignable to type 'any[][] | ((this: FieldDropdown) => any[][])'.
-    this.menuGenerator_ = FieldVariable.dropdownCreate as AnyDuringMigration;
+    this.menuGenerator_ = FieldVariable.dropdownCreate as MenuGenerator;
```

Comment:
Mmm this is kind of unfortunate (having to cast with `as`), but I think this is the right thing to do here, b/c sadly expecting a `FieldVariable` means it isn't actually a subtype :/

Reply:
Yeah, that was my thought as well. I explored a few different options, like updating `dropdownCreate` to accept generics, though in the end this made the most sense.

Ideally this could be refactored such that `this` isn't a parameter for the base function, though my goal was to avoid impacting the usage of each function.